### PR TITLE
schedule code clean: using string prefix instead of MarkDeprecated

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
schedule code clean: using string prefix instead of MarkDeprecated

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kube-scheduler/app/options/deprecated.go [hunk 1]
```diff
@@ -63,9 +62,11 @@ func (o *DeprecatedOptions) AddFlags(fs *pflag.FlagSet, cfg *kubeschedulerconfig
 	fs.StringVar(&cfg.LeaderElection.ResourceName, "lock-object-name", cfg.LeaderElection.ResourceName, "DEPRECATED: define the name of the lock object. Will be removed in favor of leader-elect-resource-name")
 
 	fs.Int32Var(&cfg.HardPodAffinitySymmetricWeight, "hard-pod-affinity-symmetric-weight", cfg.HardPodAffinitySymmetricWeight,
-		"RequiredDuringScheduling affinity is not symmetric, but there is an implicit PreferredDuringScheduling affinity rule corresponding "+
-			"to every RequiredDuringScheduling affinity rule. --hard-pod-affinity-symmetric-weight represents the weight of implicit PreferredDuringScheduling affinity rule. Must be in the range 0-100.")
-	fs.MarkDeprecated("hard-pod-affinity-symmetric-weight", "This option was moved to the policy configuration file")
+		"DEPRECATED: RequiredDuringScheduling affinity is not symmetric, but there is an implicit PreferredDuringScheduling affinity rule corresponding "+
```
Comment (draveness) [inline cmd/kube-scheduler/app/options/deprecated.go]: Do we have a precedent on deprecating arguments like this?

File: cmd/kube-scheduler/app/config/config.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ import (
 
 // Config has all the context to run a Scheduler
 type Config struct {
-	// config is the scheduler server's configuration object.
+	// ComponentConfig is the scheduler server's configuration object.
```
Comment (draveness) [inline cmd/kube-scheduler/app/config/config.go]: Good catch, could you fix golint error in this package with this change?

**Review by draveness — CHANGES_REQUESTED** [review]

File: cmd/kube-scheduler/app/options/deprecated.go (hunk 1 above)
Comment (SataQiu) [reply cmd/kube-scheduler/app/options/deprecated.go]: Yes, we've already done like this. Please see: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-scheduler/app/options/insecure_serving.go#L47-L52

we should unify deprecation mechanism, as `MarkDeprecated` hides the flag, we prefer to use string prefix `DEPRECATED:`

File: cmd/kube-scheduler/app/config/config.go (hunk 2 above)
Comment (SataQiu) [reply cmd/kube-scheduler/app/config/config.go]: Thanks! At the beginning, I thought this was a small modification, so I submitted it together.
However, if I fix golint error for the whole package, I think it is beyond the scope of this PR :(
Would you mind if I send a PR separately to fix golint error?

**Comment by SataQiu** [comment]
/assign @Huang-Wei 
PTAL :)

**Review by Huang-Wei — COMMENTED** [review]
/lgtm
/approve

Thanks, @SataQiu .

**Pull request merged**